packages/contracts/contracts/Unirep.sol

```diff
@@ -257,7 +257,8 @@ contract Unirep is IUnirep, VerifySignature {
             uint currentEpoch = updateEpochIfNeeded(uint160(msg.sender));
             if (epoch != currentEpoch) revert EpochNotMatch();
         }
-        if (epochKey >= SNARK_SCALAR_FIELD) revert InvalidEpochKey();
+        if (epochKey >= SNARK_SCALAR_FIELD || epochKey == 0 || epochKey == 1)
```

Comment:
The epoch key 1 is valid, it should instead be `SNARK_SCALAR_FIELD - 1` that is an invalid epoch key.

Reply:
if the epoch key is `1` it will be caught in 
https://github.com/Unirep/Unirep/blob/a3e7510c8c369ebfddbefd66f6a4c5eff757de7d/packages/circuits/circuits/buildOrderedTree.circom#L28-L30

Then it will be recognized as the last leaf and not processing all the other pre-images

Why is `SNARK_SCALAR_FIELD - 1` invalid?

Reply:
You're right, i was thinking about the boundaries of the array. `0` and `1` is correct.